### PR TITLE
Multiple changes in the build process

### PR DESCRIPTION
**Pull request by wonder-mice**
Fixed comments from previous pull request. CMake version is back to 2.8.

### CODE REVIEW

**Comment by warmcat** [comment]
Urgh... failed to git add lib/header.c... built here okay but broken in git, sorry about that.

Applied these thanks.

Windows installer blows up

le: "README.build.md" [compress] 3765/8706 bytes
File: "README.coding.md" [compress] 4252/10424 bytes
File: "README.test-apps.md" [compress] 3690/9384 bytes
File: "..\build\bin\Release\libwebsockets-test-client.exe" -> no files found.
Usage: File [/nonfatal] [/a] ([/r] [/x filespec [...]] filespec [...] |
   /oname=outfile one_file_only)
Error in script "libwebsockets.nsi" on line 42 -- aborting creation process
Command exited with code 1

Did I miss a patch or we broke something?

**Comment by wonder-mice** [comment]
> Did I miss a patch or we broke something?
I will investigate that in few hours. AFAIK all patches are there.

**Comment by wonder-mice** [comment]
Failed environment: LWS_METHOD=noclient, CMAKE_ARGS=-DLWS_WITHOUT_CLIENT=ON

It sort of make sense, that libwebsockets-test-client.exe was not found...

File: win32port/libwebsockets.nsi [hunk 1]
```diff
@@ -45,7 +45,7 @@ Section "Files" SecInstall
 	File "..\build\bin\Release\libwebsockets-test-ping.exe"
```
Comment (JoakimSoderberg) [inline win32port/libwebsockets.nsi]: Add `/nonfatal` to all the above exes, since they are optional to build.

Comment (wonder-mice) [inline win32port/libwebsockets.nsi]: Yep, already did.

**Comment by wonder-mice** [comment]
All tests are good.

**Pull request closed without merging**